src/test/groovy/hudson/plugins/gradle/BuildScanLogScannerTest.groovy

```diff
@@ -0,0 +1,53 @@
+package hudson.plugins.gradle
+
+
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+@Unroll
+@Subject(BuildScanLogScanner)
+class BuildScanLogScannerTest extends Specification {
+
+    def 'properly captures build scan ur given #log'(List<String> log, List<String> expectedUrls) {
+        given:
+        def listener = new MockBuildScanPublishedListener();
+        def scanner = new BuildScanLogScanner(listener)
+
+        when:
+        log.each { scanner.scanLine(it) }
+
+        then:
+        listener.urls == expectedUrls
+
+        where:
+        log                                                                                                         || expectedUrls
+        logWithBuildScans(["https://scans.gradle.com/s/bzb4vn64kx3bc"])                                             || ["https://scans.gradle.com/s/bzb4vn64kx3bc"]
+        logWithBuildScans(["https://scans.gradle.com/s/bzb4vn64kx3bc", "https://scans.gradle.com/s/asc9wm73ly1do"]) || ["https://scans.gradle.com/s/bzb4vn64kx3bc", "https://scans.gradle.com/s/asc9wm73ly1do"]
+        logWithBuildScans(["https://scans.gradle.com/bzb4vn64kx3bc"])                                               || []
+        logWithBuildScans(["https://scans.gradle.com/bzb4vn64kx3bc"], 1010)                                         || []
```

Comment:
if the goal of this is to test that we don't grab URLS after LINES_TO_SCAN, then we should use a valid URL ("https://scans.gradle.com/s/bzb4vn64kx3bc") (with the /s/ part) otherwise it doesn't prove anything